File: apps/frontend/test-api.js
#!/usr/bin/env node

/**
 * 前端功能测试脚本
 * 测试前端应用是否能正确与后端API通信
 */

const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000';

async function testHealth() {
  console.log('\n🔍 测试 1: 健康检查端点');
  try {
    const response = await fetch(`${API_BASE_URL}/health`);
    const data = await response.json();
    console.log('✅ 健康检查成功:', data);
    return true;
  } catch (error) {
    console.error('❌ 健康检查失败:', error.message);
    return false;
  }
}

async function testTools() {
  console.log('\n🔍 测试 2: 获取工具列表');
  try {
    const response = await fetch(`${API_BASE_URL}/v1/tools`);
    const data = await response.json();
    console.log('✅ 工具列表获取成功:', `找到 ${data.tools?.length || 0} 个工具`);
    if (data.tools?.length > 0) {
      console.log('   工具:', data.tools.map(t => t.name).join(', '));
    }
    return true;
  } catch (error) {
    console.error('❌ 获取工具列表失败:', error.message);
    return false;
  }
}

async function testChatReset() {
  console.log('\n🔍 测试 3: 重置对话');
  try {
    const response = await fetch(`${API_BASE_URL}/v1/chat/reset`, {
      method: 'POST',
    });
    const data = await response.json();
    console.log('✅ 重置对话成功:', data);
    return true;
  } catch (error) {
    console.error('❌ 重置对话失败:', error.message);
    return false;
  }
}

async function testHistory() {
  console.log('\n🔍 测试 4: 获取历史记录');
  try {
    const response = await fetch(`${API_BASE_URL}/v1/history?limit=10`);
    const data = await response.json();
    console.log('✅ 历史记录获取成功:', `共 ${data.total} 条消息`);
    return true;
  } catch (error) {
    console.error('❌ 获取历史记录失败:', error.message);
    return false;
  }
}

async function runTests() {
  console.log('=================================');
  console.log('  Kortix 前端功能测试');
  console.log('=================================');
  console.log(`API 地址: ${API_BASE_URL}`);

  const results = [];

  results.push(await testHealth());
  results.push(await testTools());
  results.push(await testChatReset());
  results.push(await testHistory());

  console.log('\n=================================');
  console.log('  测试结果总结');
  console.log('=================================');

  const passed = results.filter(r => r).length;
  const total = results.length;

  console.log(`✅ 通过: ${passed}/${total}`);
  console.log(`❌ 失败: ${total - passed}/${total}`);

  if (passed === total) {
    console.log('\n🎉 所有测试通过！前端应用可以正常使用');
    process.exit(0);
  } else {
    console.log('\n⚠️  部分测试失败，请检查后端服务是否正常运行');
    process.exit(1);
  }
}

runTests().catch(error => {
  console.error('\n❌ 测试执行出错:', error);
  process.exit(1);
});

File: test_api.py
#!/usr/bin/env python3
"""
API 快速测试脚本

验证所有主要端点是否正常工作
"""

import asyncio
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent))

from fastapi.testclient import TestClient

# 导入主应用（api.py 文件）
import api as api_module
app = api_module.app


def test_health():
    """测试健康检查"""
    client = TestClient(app)
    response = client.get("/health")
    print(f"✓ 健康检查: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  状态: {data['status']}")
        print(f"  工具数: {len(data['tools_enabled'])}")
        return True
    return False


def test_root():
    """测试根路径"""
    client = TestClient(app)
    response = client.get("/")
    print(f"✓ 根路径: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  名称: {data['name']}")
        print(f"  版本: {data['version']}")
        return True
    return False


def test_tools_list():
    """测试工具列表"""
    client = TestClient(app)
    response = client.get("/v1/tools")
    print(f"✓ 工具列表: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  工具数: {len(data['tools'])}")
        for tool in data['tools'][:3]:
            print(f"  - {tool['name']}: {len(tool['functions'])} 个函数")
        return True
    return False


def test_chat_completion():
    """测试非流式对话"""
    client = TestClient(app)
    response = client.post(
        "/v1/chat/completion",
        json={"message": "你好，简单介绍一下你自己"}
    )
    print(f"✓ 非流式对话: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  回复长度: {len(data['content'])} 字符")
        print(f"  预览: {data['content'][:50]}...")
        return True
    return False


def test_history():
    """测试历史接口"""
    client = TestClient(app)
    response = client.get("/v1/history")
    print(f"✓ 对话历史: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  消息数: {data['total']}")
        return True
    return False


def test_reset():
    """测试重置"""
    client = TestClient(app)
    response = client.post("/v1/chat/reset")
    print(f"✓ 重置对话: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"  成功: {data['success']}")
        return True
    return False


def main():
    """运行所有测试"""
    print("=" * 60)
    print("🧪 FastAPI 后端测试")
    print("=" * 60)
    print()
    
    tests = [
        ("健康检查", test_health),
        ("根路径", test_root),
        ("工具列表", test_tools_list),
        ("对话历史", test_history),
        ("重置对话", test_reset),
        ("非流式对话", test_chat_completion),  # 最后测试，因为会调用 LLM
    ]
    
    results = []
    for name, test_func in tests:
        try:
            result = test_func()
            results.append((name, result))
            print()
        except Exception as e:
            print(f"  ❌ 错误: {e}")
            results.append((name, False))
            print()
    
    print("=" * 60)
    print("📊 测试结果")
    print("=" * 60)
    passed = sum(1 for _, r in results if r)
    total = len(results)
    
    for name, result in results:
        status = "✅" if result else "❌"
        print(f"{status} {name}")
    
    print()
    print(f"通过: {passed}/{total}")
    
    if passed == total:
        print("\n🎉 所有测试通过！")
        return 0
    else:
        print(f"\n⚠️  {total - passed} 个测试失败")
        return 1


if __name__ == "__main__":
    sys.exit(main())
